source/app/controllers/urls_controller.rb

```diff
@@ -1,2 +1,37 @@
 class UrlsController < ApplicationController
+
+  # GET /urls
+  def index
+    @urls = Url.all
+    @url = Url.new
+  end
+
+  # POST /urls
+  def create
+    @url = Url.new(url_params)
+
+    respond_to do |format|
+      if @url.save
+        format.html { redirect_to root_path, notice: 'Url was successfully created.' }
+      else
+        format.html { redirect_to root_path, notice: @url.errors.full_messages }
+      end
+    end
+  end
+
+  # GET /:shortened_url
+  def redirect
+    @url = Url.where(shortened_url: params[:shortened_url]).take
+    if @url != nil
```

Comment:
Ruby provides [`nil?`](http://ruby-doc.org/core-2.2.2/Object.html#method-i-nil-3F) on every object for just this sort of check.

But, since you're checking for "not `nil`" the Rails method [`present?`](http://api.rubyonrails.org/classes/Object.html#method-i-present-3F) would do the trick.

Even better (in my opinion) is to drop the `nil` or presence check and depend of the fact that `nil` is falsey. Just: `if @url`.